agent.go

```diff
@@ -480,6 +502,70 @@ func (p *pod) closeChannels() {
 	}
 }
 
+// listenToUdevEvents listens to udev events to detect device nodes added
+// and binds the device nodes to the /dev of all the containers.
+// This is required as a container has its /dev mounted with tmpfs with an initial
+// list of devices.
+func (p *pod) listenToUdevEvents(done chan struct{}) {
+	u := goudev.Udev{}
+
+	// Create a monitor listening on a NetLink socket.
+	monitor := u.NewMonitorFromNetlink("udev")
+
+	// Start monitor goroutine.
+	ch, _ := monitor.DeviceChan(done)
+
+	go func() {
+		fieldLogger := agentLog.WithField("subsystem", "udevlistener")
+
+		fieldLogger.Info("Started listening for all udev events")
+
+		for d := range ch {
+			fieldLogger.WithFields(logrus.Fields{
+				"udev-path":  d.Syspath(),
+				"udev-event": d.Action(),
+			}).Info("Received udev event")
+
+			// Ignore udev events for block devices, these are handled by another go-routine
+			if d.Subsystem() == "block" {
+				continue
+			}
+
+			if isBlacklistedSubsystem(d.Subsystem()) {
```

Comment:
Thanks! Just one other comment - it might be useful to add a log call when a device is being black-listed.